runtime/near-vm-logic/src/logic.rs

```diff
@@ -637,6 +639,33 @@ impl<'a> VMLogic<'a> {
         Ok(self.context.block_index)
     }
 
+    /// Gets a block given a height (up to 256)
+    ///
+    /// # Cost
+    ///
+    /// `base`
+    // TODO #1903 rename to `block_height`
+    pub fn get_block_by_hash(&mut self, height_number: BlockHeight) -> Result<u64> {
+        self.gas_counter.pay_base(base)?;
+        let chain_store = self.context.chain_store;
+        let tip = chain_store.header_head().expect("handle this properly");
+        if tip.height == height_number {
+            return Ok(1);
+        }
+        let mut previous_block_hash = tip.prev_block_hash;
```

Comment:
Might also check `tip.height > height_number` to short-circuit impossible height and `tip.height <= height_number + 256` to short-circuit unsupported height.

Reply:
would the second inequality work given that heights aren't continuous?

Reply:
That’s the question regarding what the interface of the function is.  Does it return hashes for the last 256 blocks or for the last 256 height.  Based on https://github.com/near/NEPs/pull/384#discussion_r994284652 we want the latter in which case thee inequality should work (modulo me making mistake in whether it’s strict inequality or not).

Reply:
From what I can understand, the idea is to mimic [blockhash from ethereum](https://docs.soliditylang.org/en/v0.8.15/units-and-global-variables.html#block-and-transaction-properties). That is, it returns a block hash based on a block height searching on the latest 256 blocks.
In that case, to me the condition(s) is/are:

1. `if height > tip` -> short-circuit returns `None`
2. if height < tip - 256 we cannot short-circuit since block heights aren't continuous right?

What do you think?

Reply:
This needs to be discussed in the NEP.  Like I’ve said, https://github.com/near/NEPs/pull/384#discussion_r994284652 contradicts your interpretation.

Reply:
Added a comment there, where I hope it clarifies the behavior